rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/60_dense_vector_dynamic_mapping.yml

```diff
@@ -0,0 +1,65 @@
+          ]
+  - do:
+      indices.get_mapping:
+        index: test-too-big-still-float
+
+  - match: { test-too-big-still-float.mappings.properties.my_field.type: float }
+
+
+---
+"Fields with float arrays within the threshold map as dense_vector":
+
+  - do:
+      index:
+        index: test-map-dense-vector
+        id: "1"
+        refresh: true
+        body:
+          my_field: [
+            159.1, 289.56, -128.7424, 145.9871, -164.0003, 86.4034, -89.6929, 257.9717, 131.6075, 67.9233, -144.8255, 223.8446, 77.3228, -210.1163, -139.4783, 12.6499, 15.4491, 108.3465, -189.3947, 178.2045, -187.5925, 184.5089, 77.3022, -202.7439, -13.4959, 115.9719, -139.4332, 196.7845, 104.7573, -156.7746, 166.9878, 68.3936, 159.8473, -141.4446, 21.1947, 186.5908, -209.6895, 68.6169, 44.1255, 147.4659, 56.5079, -179.7997, -85.1651, 11.4847, 124.1662, 96.2246, -178.6705, 85.5925, 205.3616, -16.4704, 172.4947, -115.2535, -58.1722, 94.4836, 34.6458, -70.1011, -58.8047, 149.9562, -37.8998, 196.9805, -169.3555, -163.9432, 188.5611, 214.8378, 29.3182, -24.8724, 152.9382, -109.4345, -123.6716, -8.2441, 64.5902, 27.8083, 40.8185, -94.3161, 58.1463, -138.7432, 24.6805, -88.7222, -11.2018, 206.6434, 201.9024, 87.3079, -3.2883, -60.2484, -109.5789, 105.5766, -116.6709, -17.7073, -71.5093, -75.2937, -176.8691, -146.4967, 53.7586, 199.5294, 55.9754, -48.7399, 82.2051, 135.2921, 22.4408, -116.4008, -33.7538, 29.7207, 6.3692, -97.5768, -12.7982, -200.9331, -62.2743, 81.0843, 136.2247, 150.2565, 139.6838, 155.2657, -25.7447, 198.5955, 18.8099, 46.9014, -60.2672, 136.4801, 171.8966, 172.5842, 13.9123, 75.8386, -64.2444, -48.1964, 135.9685, 7.4927, -40.6424, -76.8922
+          ]
+
+
+  - do:
+      indices.get_mapping:
+        index: test-map-dense-vector
+
+  - match: { test-map-dense-vector.mappings.properties.my_field.type: dense_vector }
+  - match: { test-map-dense-vector.mappings.properties.my_field.dims: 128 }
+  - match: { test-map-dense-vector.mappings.properties.my_field.index: true }
+  - match: { test-map-dense-vector.mappings.properties.my_field.similarity: cosine }
+
```

Comment:
Could you add tests that exercise:
 - If somebody creates `dense_vector` but the `dims` don't match, we error (they specifically set the `dims` but input is incorrectly sized).
 - If somebody specifically creates a `float` mapping, we still index as `float` (this obviously doesn't touch the dynamic mapping path, just being paranoid)
 - If somebody uses a `dynamic_template` for `float`, we don't create the field as `dense_vector`.
 - If somebody uses a `dynamic_template` for `dense_vector` and the number of `float` is too few, we still create the `dense_vector` field (as long as their `dims` match the input dims size).
 - If somebody uses a `dynamic_template` for `dense_vector` but their `dims` size is WAY too big, we still throw the same mapping error (I assume we did before, I hope we did :/)

Reply:
> If somebody specifically creates a float mapping, we still index as float (this obviously doesn't touch the dynamic mapping path, just being paranoid)

I think that does on the dynamic template side. If the float field is created from the template definition:
```
"float_fields": {
          "match":   ["my_float_fields_*"],
          "mapping": {
            "type": "float"
          }
        }
```
I guess it should be treated as such and never rewritten? It's an edge case since I'd expect that arrays of more than 128 floats are not very frequent but can be confusing. Maybe we can exclude all fields created by a template explicitly and only target dynamic updates coming from our default rules using numeric detection?

Reply:
> I guess it should be treated as such and never rewritten? It's an edge case since I'd expect that arrays of more than 128 floats are not very frequent but can be confusing.

I know its an edge case, but we should make sure we don't change the mapping to `dense_vector` if the user through some logic requested the field to specifically be `float` (or `double` or anything).

Reply:
Hmm, you're right, the additional tests did expose an edge case that dynamic templates specifying float fields will still map float arrays of length between 128 and 2048 to `dense_vector`. 

I'm able to disable _all_ fields from being dynamically set to dense_vector if there's a dynamic template but that feels too strict. Is there metadata on the mapper that says it was created by a dynamic template that I'm missing? 

Either way I'll push the updates to the yaml tests, with one broken test per this conversation. Thank you!

Reply:
Thanks for the catch on the dynamic templates edge case - I've added a fix for this in addition to the requested additional test cases 👍